src/bors/mod.rs

```diff
@@ -1,87 +1,48 @@
 
 use arc_swap::ArcSwap;
-use axum::async_trait;
+use octocrab::Octocrab;
 
-use crate::bors::event::PullRequestComment;
-use crate::config::RepositoryConfig;
-use crate::database::RunId;
-use crate::github::{CommitSha, GithubRepoName, MergeError, PullRequest, PullRequestNumber};
-use crate::permissions::UserPermissions;
-use crate::TeamApiClient;
 pub use command::CommandParser;
 pub use comment::Comment;
 pub use context::BorsContext;
+#[cfg(test)]
+pub use handlers::WAIT_FOR_REFRESH;
 pub use handlers::{handle_bors_global_event, handle_bors_repository_event};
 
-/// Provides functionality for working with a remote repository.
-pub trait RepositoryClient: Send + Sync {
-    fn repository(&self) -> &GithubRepoName;
-
-    /// Was the comment created by the bot?
-    async fn is_comment_internal(&self, comment: &PullRequestComment) -> anyhow::Result<bool>;
-
-    /// Load repository config.
-    async fn load_config(&self) -> anyhow::Result<RepositoryConfig>;
-
-    /// Return the current SHA of the given branch.
-    async fn get_branch_sha(&self, name: &str) -> anyhow::Result<CommitSha>;
-
-    /// Resolve a pull request from this repository by it's number.
-    async fn get_pull_request(&self, pr: PullRequestNumber) -> anyhow::Result<PullRequest>;
-
-    /// Post a comment to the pull request with the given number.
-    async fn post_comment(&self, pr: PullRequestNumber, comment: Comment) -> anyhow::Result<()>;
-
-    /// Set the given branch to a commit with the given `sha`.
-    async fn set_branch_to_sha(&self, branch: &str, sha: &CommitSha) -> anyhow::Result<()>;
-
-    /// Merge `head` into `base`. Returns the SHA of the merge commit.
-    async fn merge_branches(
-        &self,
-        base: &str,
-        head: &CommitSha,
-        commit_message: &str,
-    ) -> Result<CommitSha, MergeError>;
-
-    /// Find all check suites attached to the given commit and branch.
-    async fn get_check_suites_for_commit(
-        &self,
-        branch: &str,
-        sha: &CommitSha,
-    ) -> anyhow::Result<Vec<CheckSuite>>;
-
-    /// Cancels Github Actions workflows.
-    async fn cancel_workflows(&self, run_ids: &[RunId]) -> anyhow::Result<()>;
-
-    /// Add a set of labels to a PR.
-    async fn add_labels(&self, pr: PullRequestNumber, labels: &[String]) -> anyhow::Result<()>;
+use crate::config::RepositoryConfig;
+use crate::github::api::client::GithubRepositoryClient;
+use crate::github::api::load_repositories;
+use crate::github::GithubRepoName;
+use crate::permissions::UserPermissions;
+use crate::TeamApiClient;
 
-    /// Remove a set of labels from a PR.
-    async fn remove_labels(&self, pr: PullRequestNumber, labels: &[String]) -> anyhow::Result<()>;
+mod command;
+pub mod comment;
+mod context;
+pub mod event;
+mod handlers;
 
-    /// Get a workflow url.
-    fn get_workflow_url(&self, run_id: RunId) -> String;
+/// Loads repositories through a global GitHub client.
+pub struct RepositoryLoader {
+    client: Octocrab,
```

Comment:
I think we could remove this struct and just directly call load_repositories?

Reply:
I started with that, but I decided to keep the struct in the end. Because we use a bunch of Octocrab clients in the app, but only the global one is usable for loading the repositories, and it's easy to accidentally mix them up. So this is now essentially just a typed wrapper to make it harder to use the wrong client for loading the repositories.

Reply:
I see. If that's the case, do you think RepositoryLoader and TeamApiClient should be combined? They are always used together to load/reload repositories, and like you said, both also use a global octocrab client

Reply:
TeamApiClient doesn't use Octocrab, it doesn't communicate with GitHub :) But even then, I wouldn't combine them, they have different responsibilities.

Reply:
Hmmm, still, I don't really love this 
```rust
repo_loader.load_repositories(team_api_client).await
```
There should be other way to do this, right?

Reply:
I don't know, is it that bad? Seems like normal dependency injection to me, the method needs a dependency, so we provide the dependency to it.

Btw, I probably misunderstood your "should be combined" suggestion. I thought that you meant to literally combine these two structs into one (which I wouldn't do), but you probably just meant to add `TeamApiClient` as a field into `RepositoryLoader`? We could do that, although then we'd need to either clone the client or put it behin a shared pointer, and I don't think doing that provides enough benefit here vs just passing it as an argument.

Reply:
> I don't know, is it that bad? Seems like normal dependency injection to me, the method needs a dependency, so we provide the dependency to it.

Yeah, I mean, it feels kinda awkward. Like, RepoLoader, load repositories for me, and then there is this TeamApiClient. But I'm just being nitpicky here.

>  We could do that, although then we'd need to either clone the client or put it behind a Rc pointer, and I don't think doing that provides enough benefit here vs just passing it as an argument.

Yeah it sounds more hassle than necessary. Let's just keep it that way.